clr-k8s-examples/vagrant.md

```diff
@@ -0,0 +1,29 @@
+# Detailed Vagrant installation steps
+## Install vagrant on Ubuntu
+
+On Ubuntu Bionic, run these commands
+Install dependencies and prepare system
+```bash
+sudo apt-get update
+sudo apt-get install gcc make
+sudo apt-get install qemu qemu-kvm libvirt-bin ebtables dnsmasq-base virt-top  libguestfs-tools virtinst bridge-utils
+sudo apt-get install libxslt-dev libxml2-dev libvirt-dev zlib1g-dev ruby-dev
+sudo modprobe vhost_net
+sudo lsmod | grep vhost
+echo "vhost_net" | sudo tee -a /etc/modules
+```
+Download the latest Debian package from https://www.vagrantup.com/downloads.html and install it followed by vagrant-libvirt
+```bash
+sudo dpkg -i vagrant_${VER}_x86_64.deb
+sudo vagrant plugin install vagrant-libvirt
+```
+Run vagrant
+```bash
+sudo vagrant up --provider=libvirt
```

Comment:
Why does this guide run vagrant as root?

The main `README.md` doesn't and then doesn't find the libvirt provider.

I tried this on Debian Buster as normal user, but it didn't work for me. :man_shrugging: 

```
$ vagrant plugin install vagrant-libvirt
Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Fetching: formatador-0.2.5.gem (100%)
Fetching: excon-0.66.0.gem (100%)
Fetching: fog-core-1.43.0.gem (100%)
Fetching: fog-json-1.2.0.gem (100%)
Fetching: mini_portile2-2.4.0.gem (100%)
Fetching: nokogiri-1.10.4.gem (100%)
Building native extensions.  This could take a while...
Fetching: fog-xml-0.1.3.gem (100%)
Fetching: ruby-libvirt-0.7.1.gem (100%)
Building native extensions.  This could take a while...
Fetching: fog-libvirt-0.6.0.gem (100%)
Fetching: vagrant-libvirt-0.0.45.gem (100%)
Installed the plugin 'vagrant-libvirt (0.0.45)'!

$ NODES=1 MEMORY=8192 CPUS=8 vagrant up --provider=libvirt
Bringing machine 'default' up with 'libvirt' provider...
==> default: Box 'base' could not be found. Attempting to find and install...
    default: Box Provider: libvirt
    default: Box Version: >= 0
==> default: Box file was not detected as metadata. Adding it directly...
==> default: Adding box 'base' (v0) for provider: libvirt
    default: Downloading: base
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

Couldn't open file /home/pohly/base
```

Reply:
The "Couldn't open file /home/pohly/base" error was my fault, I ran the command in the root of the repo instead of `clr-k8s-examples`.

`--provider=libvirt` seems to work now for me when using the "vagrant" commands without "sudo" (machines run, `swupd update` is reached), but then provisioning fails with:
```
...

    clr-03: Update was applied.
    clr-03: Calling post-update helper scripts.
    clr-03: none
    clr-01: pacdiscovery.service: restarted (the binary was updated)
    clr-02: pacdiscovery.service: restarted (the binary was updated)
    clr-03: pacdiscovery.service: restarted (the binary was updated)
    clr-01: systemd-udevd.service: restarted (the binary was updated)
    clr-02: systemd-udevd.service: restarted (the binary was updated)
    clr-01: pacrunner.service: restarted (the binary was updated)
    clr-02: pacrunner.service: restarted (the binary was updated)
    clr-03: systemd-udevd.service: restarted (the binary was updated)
    clr-03: pacrunner.service: restarted (the binary was updated)
    clr-01: systemd-journald.service: restarted (the binary was updated)
    clr-02: systemd-journald.service: restarted (the binary was updated)
    clr-01: mcelog.service: restarted (the binary was updated)
    clr-02: mcelog.service: restarted (the binary was updated)
    clr-03: systemd-journald.service: restarted (the binary was updated)
    clr-03: mcelog.service: restarted (the binary was updated)
    clr-01: systemd-resolved.service: restarted (the binary was updated)
    clr-02: systemd-resolved.service: restarted (the binary was updated)
    clr-01: systemd-timesyncd.service: restarted (the binary was updated)
    clr-02: systemd-timesyncd.service: restarted (the binary was updated)
    clr-01: Update took 423.4 seconds, 211 MB transferred
    clr-03: systemd-resolved.service: restarted (the binary was updated)
    clr-02: Update took 423.8 seconds, 211 MB transferred
    clr-03: systemd-timesyncd.service: restarted (the binary was updated)
    clr-02: 6363 files were not in a pack
    clr-02: Update successful. System updated from version 30260 to version 30760
==> clr-02: An error occurred. The error will be shown after all tasks complete.
    clr-01: 6363 files were not in a pack
    clr-01: Update successful. System updated from version 30260 to version 30760
==> clr-01: An error occurred. The error will be shown after all tasks complete.
    clr-03: Update took 424.0 seconds, 211 MB transferred
    clr-03: 6363 files were not in a pack
    clr-03: Update successful. System updated from version 30260 to version 30760
==> clr-03: An error occurred. The error will be shown after all tasks complete.
An error occurred while executing multiple actions in parallel.
Any errors that occurred are shown below.

An error occurred while executing the action on the 'clr-01'
machine. Please handle this error then try again:

The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

An error occurred while executing the action on the 'clr-02'
machine. Please handle this error then try again:

The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

An error occurred while executing the action on the 'clr-03'
machine. Please handle this error then try again:

The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

:man_shrugging: